### PR TITLE
Assert: Add `assert.closeTo()`

### PR DESCRIPTION
**Pull request by Krinkle**
Ref https://github.com/qunitjs/qunit/issues/1735

### CODE REVIEW

File: docs/api/assert/closeTo.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+| name | description |
+|------|-------------|
+| `actual` (number) | Expression being tested |
+| `expected` (number) | Known comparison value |
+| `delta` (number) |  |
+| `message` (string) | Short description of the actual expression |
+
+
+TODO: close, close-enough, roughlyEquals, closeTo, approximate
```
Comment (timmywil) [inline docs/api/assert/closeTo.md:24]: Did you mean to leave this in? I don't think it needs to be user-facing. And as a user, I don't know what this means. Are all of these methods coming? `closeTo` is part of the list. Is there more to do on that one?

Comment (Krinkle) [reply docs/api/assert/closeTo.md:24]: These are keywords I meant to incorporate in the long-form description to help with search and discovery. I thought I had finished the docs, but I must have dreamed that. Consider this markdown file incomplete :)

File: test/cli/fixtures/assert-failure.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  QUnit.test('closeTo [failure]', function (assert) {
+    assert.closeTo(1, 2, 0);
+    assert.closeTo(1, 2, 1);
+    assert.closeTo(2, 7, 1);
+
+    assert.closeTo(7, 7.3, 0.1);
+    assert.closeTo(7, 7.3, 0.2);
+    assert.closeTo(2011, 2013, 1);
```
Comment (gibson042) [inline test/cli/fixtures/assert-failure.js:23]: There should be tests for errors from missing/non-number delta; is this the right place for them?
```suggestion
    assert.closeTo(2011, 2013, 1);
    
    assert.closeTo(1, 1); // missing delta
    assert.closeTo(1, 1, false); // non-number delta
```

File: docs/api/assert/closeTo.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+groups:
+  - assert
+redirect_from:
+  - "/assert/closeTo/"
+version_added: "unreleased"
+---
+
+`closeTo( actual, expected, delta, message = "" )`
+
+Check that an object contains certain properties.
```
Comment (gibson042) [inline docs/api/assert/closeTo.md:14]:
```suggestion
excerpt: Check that a number value is within some tolerance of an expected value.
groups:
  - assert
redirect_from:
  - "/assert/closeTo/"
version_added: "unreleased"
---

`closeTo( actual, expected, delta, message = "" )`

Check that a number value is within some tolerance of an expected value.
```

File: docs/api/assert/closeTo.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+| name | description |
+|------|-------------|
+| `actual` (number) | Expression being tested |
+| `expected` (number) | Known comparison value |
+| `delta` (number) |  |
```
Comment (gibson042) [inline docs/api/assert/closeTo.md:20]:
```suggestion
| `delta` (number) | Maximum allowable difference between actual and expected |
```

File: docs/api/assert/closeTo.md (hunk 1 above)
Comment (Krinkle) [reply docs/api/assert/closeTo.md]: Fixed.

File: test/cli/fixtures/assert-failure.js (hunk 2 above)
Comment (Krinkle) [reply test/cli/fixtures/assert-failure.js:23]: Yep. Done!

**Review by gibson042 — APPROVED** [review]

**Pull request merged**